# num 24

##### № 25361 ЕГКР 13.12.25 (Уровень: Базовый)

![alt text](tasks/{40817934-8706-4BDD-8485-8FF5456F1C07}.png)

In [ ]:
from string import printable


chets = printable[2:10:2]
max_len = 0
now_len = 0
cnt_f = 0
string = open("24_25361.txt").readline()
for char in string:
    if char in chets:
        if cnt_f == 76:
            if max_len < now_len:
                max_len = now_len
        cnt_f = 0
        now_len = 1
    elif char == "F": 
        if now_len != 0:
            cnt_f += 1
            now_len += 1
            if cnt_f >= 76:
                if now_len > max_len:
                    max_len = now_len
                    now_len = 0
                cnt_f = 0
    else:
        if now_len != 0:
            now_len += 1
print(max_len)

Ответ: 163

##### № 28765 Досрочная волна 2026 (Уровень: Базовый)

Текстовый файл состоит из заглавных букв латинского алфавита A, B, C, D, E и F.
Определите максимальное количество идущих подряд символов в прилагаемом файле, среди которых пара символов BС (в указанном порядке) встречается не более 180 раз.

In [ ]:
string = open("tasks/24_28765.txt").readline().replace("BC", "B C").split(" ")
max_len = 0
st = 0
for _ in range(len(string) - 180):
    cur_len = 0
    for i in range(st, st + 181):
        cur_len += len(string[i])
    if cur_len > max_len:
        max_len = cur_len
    st += 1
    cur_len = 0
print(max_len)

38442


Ответ: 38442

##### № 27777 Апробация 04.03.26 (Уровень: Базовый)

Текстовый файл состоит из символов, обозначающих заглавные буквы латинского алфавита и цифры от 1 до 9 включительно.
Определите в прилагаемом файле максимальное количество идущих подряд символов, которые могут представлять запись числа в 
двенадцатеричной системе счисления.

Примечание. Цифры, числовое значение которых превышает 9, обозначены латинскими буквами начиная с буквы А.

In [ ]:
import re

string = open("tasks/24_27777.txt").readline()
max_len = 0

for match in re.finditer(r"(\d|A|B)+", string):
    if (maybe_max_len := len(match.group())) > max_len:
        max_len = maybe_max_len

print(max_len)

Ответ: 18

##### № 27634 Апробация 04.03.26 (Уровень: Базовый)

Текстовый файл состоит из символов T, U, V, W, X, Y и Z.

Определите в прилагаемом файле минимальное количество идущих подряд символов (длину непрерывной подпоследовательности), среди которых символ Z встречается не менее 270 раз.

In [ ]:
string = open("tasks/24_27634.txt").readline().replace("Z", "Z ").split()
min_len = float("inf")

for a in range(len(string) - 270):
    cur_len = 1
    cur_len += sum(map(len, string[a + 1:a + 270]))
    if cur_len < min_len:
        min_len = cur_len

print(min_len)

Ответ: 1058